drivers/adc/ad469x/ad469x.c

```diff
@@ -803,11 +903,11 @@ int32_t ad469x_init(struct ad469x_dev **device,
 	}
 #endif
 
-	ret = ad469x_init_gpio(dev, init_param);
+	ret = no_os_spi_init(&dev->spi_desc, init_param->spi_init);
```

Comment:
Actually, before creating a  ad469x_reset_dev() function, it was contained inside the ad469x_init_gpio() which would call the spi_read function read the status registers after reset. Since spi_init happens after it, so it shall result into a hard fault.
 
But now it is different function and I have updated the code accordingly.